client/sttp-client4/src/main/scala/sttp/tapir/client/sttp4/SttpClientInterpreter.scala

```diff
@@ -0,0 +1,228 @@
+      wsToPipe: WebSocketToPipe[R]
+  ): I => F[DecodeResult[Either[E, O]]] = { (i: I) =>
+    val req = toRequest(e, baseUri)
+    backend.monad.map(GenericRequestExtensions.sendRequest(backend, req(i)))(_.body)
+  }
+
+  /** Interprets the public endpoint as a client call, using the given `baseUri` as the starting point to create the target uri. If
+    * `baseUri` is not provided, the request will be a relative one.
+    *
+    * Returns a function which, when applied to the endpoint's input parameters (given as a tuple), will encode them to appropriate request
+    * parameters: path, query, headers and body. The request is sent using the given backend, and the result (error or success value) is
+    * returned. If decoding the result fails, a failed effect is returned instead.
+    */
+  def toClientThrowDecodeFailures[F[_], I, E, O, R](e: PublicEndpoint[I, E, O, R], baseUri: Option[Uri], backend: Backend[F])(
+      implicit wsToPipe: WebSocketToPipe[R]
+  ): I => F[Either[E, O]] = {
+    val req = toRequestThrowDecodeFailures(e, baseUri)
+    (i: I) => backend.monad.map(GenericRequestExtensions.sendRequest(backend, req(i)))(_.body)
+  }
+
+  /** Interprets the public endpoint as a client call, using the given `baseUri` as the starting point to create the target uri. If
+    * `baseUri` is not provided, the request will be a relative one.
+    *
+    * Returns a function which, when applied to the endpoint's input parameters (given as a tuple), will encode them to appropriate request
+    * parameters: path, query, headers and body. The request is sent using the given backend, and the result (success value) is returned. If
+    * decoding the result fails, or if the response corresponds to an error value, a failed effect is returned instead.
+    */
+  def toClientThrowErrors[F[_], I, E, O, R](e: PublicEndpoint[I, E, O, R], baseUri: Option[Uri], backend: Backend[F])(implicit
+      wsToPipe: WebSocketToPipe[R]
+  ): I => F[O] = {
+    val req = toRequestThrowErrors(e, baseUri)
+    (i: I) => backend.monad.map(GenericRequestExtensions.sendRequest(backend, req(i)))(_.body)
+  }
+
+  /** Interprets the public endpoint as a client call, using the given `baseUri` as the starting point to create the target uri. If
+    * `baseUri` is not provided, the request will be a relative one.
+    *
+    * Returns a function which, when applied to the endpoint's input parameters (given as a tuple), will encode them to appropriate request
+    * parameters: path, query, headers and body. The result of the function is a description of a request, which can be sent using any sttp
+    * backend. The response will then contain the decoded error or success values (note that this can be the body enriched with data from
+    * headers/status code).
+    */
+  def toRequest[I, E, O, R](e: PublicEndpoint[I, E, O, R], baseUri: Option[Uri])(implicit
+      wsToPipe: WebSocketToPipe[R]
+  ): I => GenericRequest[DecodeResult[Either[E, O]], Any] = {
```

Comment:
here, we should return a specific `Request` subtype. `GenericRequest` is not something end-users should be working with

Reply:
fixed